commander/src/bootstrapping/commands/endpoint/list.ts

```diff
@@ -0,0 +1,94 @@
+import { Schema } from '@liskhq/lisk-codec';
+import { Flags as flagParser } from '@oclif/core';
+import { BaseIPCClientCommand } from '../base_ipc_client';
+
+export class ListCommand extends BaseIPCClientCommand {
+	static description = 'Lists registered endpoints.';
+
+	static examples = [
+		'endpoint:list',
+		'endpoint:list {endpoint} -m {module}',
+		'endpoint:list {endpoint} -m {module} -i',
+		'endpoint:list -m validator',
+		'endopint:list getBalance',
+		'endpoint:list get -m token ',
+		'endpoint:list getBalances -m token -i --pretty',
+		'endpoint:list getBalances -m token -d ~/.lisk/pos-mainchain',
+	];
+
+	static args = [
+		{
+			name: 'endpoint',
+			description: 'Endpoint name (Optional)',
+			required: false,
+		},
+	];
+
+	static flags = {
+		...BaseIPCClientCommand.flags,
+		module: flagParser.string({
+			char: 'm',
+			description: 'Parent module.',
+		}),
+		info: flagParser.boolean({
+			char: 'i',
+			description: 'Prints additional info; Request and Response objects.',
+		}),
+	};
+
+	async run(): Promise<void> {
+		const { args, flags } = await this.parse(ListCommand);
+
+		if (!this._client) {
```

Comment:
Is this logic tested? I don't see it in the unit tests?

Reply:
I have tried creating an instance, instead of triggering it through `ListCommand.run` and mocking `error` method to verify if it gets called to no avail, couldn't find any other valid approach in existing commands that extend from `BaseIPCClient`, there are quite a few and from what I have found all skip over test for this case, probably there is a reason for it? @ishantiw

Reply:
Thanks @has5aan , maybe you could try

```ts
import { homedir } from 'os';
import { join } from 'path';

const defaultDataPath = join(homedir(), '.lisk', '@oclif', 'core');

it('should throw an error if the application is not running', async () => {
	jest.spyOn(appUtils, 'isApplicationRunning').mockReturnValue(false);

	await expect(ListCommand.run([], config)).rejects.toThrow(`Application at data path ${defaultDataPath} is not running.`);
});

it('should throw an error if the ipc api client is not running', async () => {
	jest.spyOn(apiClient, 'createIPCClient').mockRejectedValue(new Error('APIClient is not initialized.'));

	await expect(ListCommand.run([], config)).rejects.toThrow('APIClient is not initialized.');
});
```

I've mocked the `appUtils, 'isApplicationRunning'` & `apiClient, 'createIPCClient'`.

Reply:
Thanks @mosmartin 🙏🏽, added both the recommended tests taking help from your examples.